presentation/src/main/java/com/example/user/test/presentation/base/BaseActivity.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.user.test.presentation.base
+
+import android.os.Bundle
+import android.os.PersistableBundle
+import android.support.v7.app.AppCompatActivity
+import android.util.Log
+import com.arellomobile.mvp.MvpAppCompatActivity
+import com.arellomobile.mvp.MvpPresenter
+import com.arellomobile.mvp.presenter.InjectPresenter
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+
+abstract class BaseActivity<P: MvpPresenter<*>, R: BaseRouter<*>>: MvpAppCompatActivity() {
+
+    protected val router: R
+
+    init {
+        router = provideRouter()
+    }
+
+    private val compositeDisposable: CompositeDisposable by lazy {
+        CompositeDisposable()
+    }
+
+    abstract fun provideLayoutId(): Int
+
+    abstract fun provideRouter(): R
```

Comment:
Refactor according to `Cicerone`. Inject `NavigationHolder` and add `abstract val Navigator` and every child will override it with `SupportAppNavigator` and correct `layoutId` into `BaseActivity`. See `94851cfc565a8144cc87d00816b149f64409fae` commit hash revision in `GitFox`.